### DISCUSSION TITLE
OpenSnitch does not start (aarch64, manjaro)

### DISCUSSION THREAD

**Discussion by Doroshenko-T** [evilsocket/opensnitch#1147] [General]
Hello friends,
Unfortunately, this wonderful program did not work for me :-(

```
$ opensnitch-ui
Traceback (most recent call last):
  File "/usr/bin/opensnitch-ui", line 36, in <module>
    import grpc
  File "/usr/lib/python3.11/site-packages/grpc/__init__.py", line 22, in <module>
    from grpc import _compression
  File "/usr/lib/python3.11/site-packages/grpc/_compression.py", line 20, in <module>
    from grpc._cython import cygrpc
ImportError: libgmock.so.1.14.0: cannot open shared object file: No such file or directory

```

```
# opensnitchd
[2024-01-06 14:14:17]  IMP  Starting opensnitch-daemon v1.6.2
[2024-01-06 14:14:17]  INF  Loading rules from /root/rules ...
[2024-01-06 14:14:17]  !!!  Path '/root/rules' does not exist
Create it in if you want to save rules to disk
```

Environment:  Manjaro XFCE ARM aarch64

**Comment by ple1n** [comment]
libgmock ? Sounds like something you can install from package manager

**Comment by Doroshenko-T** [comment]
Sorry, I'm not good at these things.
Please make a correct build OpenSnitch so that it does not create stupid errors at startup.

**Comment by gustavo-iniguez-goya** [comment]
Hi @T-Shilov ,

How did you install the package? with yay for example it would be: `yay -S opensnitch`

**Comment by Doroshenko-T** [comment]
I installed the package OpenSnitch 1.6.4-1using GUI installer.
And  this package was installed without errors.

**Comment by gustavo-iniguez-goya** [comment]
ok, so now is everything working?

**Comment by Doroshenko-T** [comment]
I have already said that OpenSnitch does not work. 
Therefore, I have uninstalled it and am waiting for somebody to build a correct build OpenSnitch for ARM  that will run without errors.

**Comment by gustavo-iniguez-goya** [comment]
I think you're waiting on the wrong forum then, because the opensnitch AUR package is not done here.
Did you try asking on https://archlinux.org/packages/extra/x86_64/opensnitch/ for arm builds?

Regarding this error:
```bash
# opensnitchd
 [2024-01-06 14:14:17]  IMP  Starting opensnitch-daemon v1.6.2
 [2024-01-06 14:14:17]  INF  Loading rules from /root/rules ...
 [2024-01-06 14:14:17]  !!!  Path '/root/rules' does not exist
 Create it in if you want to save rules to disk
```

if you create the rules dir (`mkdir -p /etc/opensnitchd/rules`) and launch it like this it should work: `opensnitchd -rules /etc/opensnitchd/rules`.
But in any case, if you're running a aarch64 system where did you get the binary for arm , from the github releases section? I thought the arch package was only for x86_64 systems.

Anyway, you'll still have to fix the GUI dependency error.

**Comment by Doroshenko-T** [comment]
![About_OpenSnitch](https://github.com/evilsocket/opensnitch/assets/61428202/0c0890c0-eec4-4706-ad07-68f4e23e1b0d)


I don't need to ask https://archlinux.org/packages/extra/x86_64/opensnitch
I am not an expert, but an ordinary user, So I am not obliged to create any rules that are incomprehensible to me. 
This should be done by the installer Open Snitch during installation. 
And if he's not doing it now, , then the application builder must fix his mistake on his own.
And only after that me can use this application.

Many other applications run without errors, without manual fixes, and work fine.
So, iIt is not up to me to fix the GUI error, but the application creator or he the builder.

I am very disappointed that OpenSnitch does not work on ARM. But It works for me on x86_64.
Can you tell me who put together such an unsuccessful OpenSnitch package for ARM?
And how can I contact him directly so that he corrects his mistake?

**Comment by b14ckw1d0w** [comment]
While you are correct that many applications run without the need for user intervention. In this case, you are the error. Do some legwork and maybe even put a few characters into Google and make those brain cells work a little harder than they currently are.

As you have been told, OpenSnitch works perfectly fine. Contact your distribution maker to find out why they have not put a working package into their repo. Archlinx.org won't help you, and rightly so. You chose an ARM version of Manjaro, which is not Arch btw. Usually, you would ask on the Arch ARM forum but even that is incorrect since Manjaro is not Arch. So go ask on the Manjaro forum why the package that THEY put together doesn't work.

Or you could email the email address that you posted in your last post, perhaps the person who put this package together for the AUR may know why it does not work, but that requires some common sense to figure that one out.

If you want to become more than just an ordinary user and learn how to install a package then I suggest you start learning before you come on here and start slating the makers of OpenSnitch!

I take it you researched whether this will work on your arm system, or do you just install things and complain to the wrong people all the time?

**Comment by Doroshenko-T** [comment]
I have already sent an email to the packer at his address:  [builder+seattle@archlinuxarm](url)
But email returned with an error. It looks like this value address is fake.

**Comment by jozeywhales** [comment]
I installed Opensnitch from AUR (Opensnitch 1.6.4-1) in Big Linux (arch/manjaro KDE). 
It was in the menu, would open but wouldn't run.
I got it running by using  ...  **sudo systemctl restart opensnitchd.service**.
It wouldn't autostart with the PC so I added it to the startup apps and now it autostarts fine after boot.

Immediately after boot I get the errors below but Opensnitch seems to running fine.
After about 30-45 seconds the error messages on the Desktop go away but don't know if that's from [eBPF] functioning or not?

![opensnitch error](https://github.com/evilsocket/opensnitch/assets/160407583/6cd3a2fb-bcbe-47d3-a0ad-28eaeaf3607d)

**Comment by ple1n** [comment]
You can usually use `systemctl enable` to make service start at boot

**Comment by jozeywhales** [comment]
> You can usually use `systemctl enable` to make service start at boot

I tried that but not working for some reason.

**Comment by atomGit** [comment]
@T-Shilov - your attitude toward developers of **free and open source software** needs a serious adjustment - you have **no right** to demand anything and **no reason** to treat developers that poorly when they've not given you any reason to do so - it's that kind of attitude that can cause a good developer to throw in the towel

[When you use open source software you are not entitled to anything!](https://unixdigest.com/articles/when-you-use-open-source-software-you-are-not-entitled-to-anything.html)

**Comment by Doroshenko-T** [comment]
@**atomGit** 
I'm sorry, I'm not demanding, but only asking you to fix some flaws.
I am very glad and grateful to the people who created this wonderful application. 
I'm happy to use the OpenSnitch on other computers, but it very upsets me if it doesn't work as well somewhere as I hoped.
Sorry, sorry, sorry...
